lib/cretonne/src/isa/intel/abi.rs

```diff
@@ -66,8 +66,7 @@ impl ArgAssigner for Args {
         }
 
         // Handle special-purpose arguments.
-        // TODO: The registers below are for `spiderwasm`. Should we check the calling convention?
-        if ty.is_int() {
+        if ty.is_int() && call_conv == ir::CallConv::SpiderWASM {
```

Comment:
With `call_conv` as an `Args` member, this then becomes `if ty.is_int() && self.call_conv == ir::CallConv::SpiderWASM`.